### ISSUE TITLE
No issue: question about faster algorithm

### ISSUE THREAD

**Issue by kurtby** [jcjohnson/neural-style#261]
Hi!

A new popular service has just launched - Prisma (https://goo.gl/nNtzDR) and they use neural networks to create filters. The results they get are very different, yet the speed is high.
They mentioned that they used the same algorithm as yours as a start. But their filters are super fast. It’s interesting how they did it. Do you you have any ideas how they did it?

**Comment by ProGamerGov** [comment]
The settings discussed in this issue here: https://github.com/jcjohnson/neural-style/issues/240 Can create the best iterations, at iteration 50. I'd assume they are doing something similar.

**Comment by 6o6o** [comment]
This is unconfirmed and my personal assumption, but they might be using https://github.com/yusuketomoto/chainer-fast-neuralstyle

I discovered it accidentally, while exploring work of Gene Kogan. It was used in [Cubist Mirror](https://vimeo.com/167910860), a project for an art exhibition. Processing happens almost realtime by doing a single forward pass, although it's much harder to obtain a descent style, since you need to train a separate network for it, which is probably why you cannot set a custom style in Prisma.

**Comment by jcjohnson** [comment]
I recently wrote a paper about using feedforward networks for real-time style transfer:

http://arxiv.org/abs/1603.08155

chainer-fast-neuralstyle is a reimplementation of my paper; I agree that Prisma is likely using this in the backend. I haven't released my code for this paper yet, but I likely will.

**Comment by 3DTOPO** [comment]
Very impressive jc! Please let us know when you do (hint hint)!!

**Comment by rayset** [comment]
@ProGamerGov what do you mean by iteration 50?

this command gives you good results at iteration 50? for me it does not, mind sharing some examples?

th neural_style3.lua -image_size 1000 -content_image content.jpg -style_image image.jpg -content_layers relu4_2 -style_layers relu1_1,relu2_1,relu3_1,relu4_1,relu5_1 -style_layer_weights 1,1,1,20,1-content_weight 100 -style_weight 1500 -init image -normalize_gradients -num_iterations 1500 -backend cudnn -cudnn_autotune -optimizer adam

congratulation to jcjohnson for the paper, it's amazing.

**Comment by ProGamerGov** [comment]
@jcjohnson I'm looking forward to potentially experimenting with that code! 


@rayset Remove the style_weights option and play around with a learning_rate value of between 1 and 50. Though really low learning rate values that are closer to 0, seem to work best.

**Comment by rayset** [comment]
@ProGamerGov the default for leraning rate is 0.001, I tried with 0.00001 but no luck, basically the same output. what values have you tried? I don't think the learning rate is supposed to be higher than 1

**Comment by wangzhangup** [comment]
@jcjohnson Very impressive!!!!!!!!!!!!!!!!!!!!!!!

**Comment by austingg** [comment]
@jcjohnson  @3DTOPO  @6o6o  @ProGamerGov  
I have found a very impressive charactersic of prisma. It can tell by the smooth area such as sky! 
![img_0593](https://cloud.githubusercontent.com/assets/3223849/17045735/bdfad75c-4ffe-11e6-8332-bc8d3f410198.JPG)

**Comment by mystcat** [comment]
It is not the Prisma's only characteristic. I found other application with the same or higher quality, like FrameAI - see example.
![b6pnpja5co6bloitxt88](https://cloud.githubusercontent.com/assets/18401347/17054548/c928f38a-4fd5-11e6-8671-9d464c3dba6f.jpg)
